### ISSUE TITLE
An exception in synchronize module when SRC or DEST args contain unicode paths

### ISSUE THREAD

**Issue by dmromanov** [ansible/ansible#14041]
##### Issue Type:

Bug Report
##### Ansible Version:

```
$ ansible --version
ansible 2.1.0 (devel 627dec716b) last updated 2016/01/21 12:46:03 (GMT +300)
  lib/ansible/modules/core: (detached HEAD d7fac82f97) last updated 2016/01/21 12:46:12 (GMT +300)
  lib/ansible/modules/extras: (detached HEAD f798240f43) last updated 2016/01/21 12:46:18 (GMT +300)
  config file = /home/dromanov/ansible-test/ansible.cfg
  configured module search path = Default w/o overrides
```
##### Ansible Configuration:

```
[defaults]
hostfile = hosts
```
##### Environment:

Ubuntu 15.10
Python 2.7.10
##### Summary:

An exception is thrown in synchronize module when SRC or DEST args contain unicode paths
##### Steps To Reproduce:

Run ansible-playbook in verbose mode `$ ansible-playbook debug.yml -vvvv`.

Project configuration:

./ansible.cfg

```
[defaults]
hostfile = hosts
```

./hosts

```
[local]
localhost ansible_connection=local
```

./debug.yml

```

---
- hosts: localhost

  tasks:
  - synchronize: src=/tmp/пример1 dest=/tmp/пример2
```
##### Expected Results:

```
TASK [synchronize] *************************************************************
task path: /home/dromanov/ansible-test/deploy.yml:5
ESTABLISH LOCAL CONNECTION FOR USER: dromanov
localhost EXEC /bin/sh -c '( umask 22 && mkdir -p "$( echo $HOME/.ansible/tmp/ansible-tmp-1453373512.16-200246025287651 )" && echo "$( echo $HOME/.ansible/tmp/ansible-tmp-1453373512.16-200246025287651 )" )'
localhost PUT /tmp/tmpEWdFnL TO /home/dromanov/.ansible/tmp/ansible-tmp-1453373512.16-200246025287651/synchronize
localhost EXEC /bin/sh -c 'LANG=ru_RU.UTF-8 LC_ALL=ru_RU.UTF-8 LC_MESSAGES=ru_RU.UTF-8 /usr/bin/python /home/dromanov/.ansible/tmp/ansible-tmp-1453373512.16-200246025287651/synchronize; rm -rf "/home/dromanov/.ansible/tmp/ansible-tmp-1453373512.16-200246025287651/" > /dev/null 2>&1'
fatal: [localhost]: FAILED! => {"changed": false, "cmd": "rsync --delay-updates -F --compress --archive --rsh 'ssh  -S none -o StrictHostKeyChecking=no' --out-format='<<CHANGED>>%i %n%L' \"/tmp/1\" \"/tmp/2\"", "failed": true, "invocation": {"module_args": {"_ansible_check_mode": false, "_ansible_debug": false, "_ansible_diff": false, "_ansible_no_log": false, "_ansible_verbosity": 4, "_local_rsync_path": "rsync", "archive": true, "checksum": false, "compress": true, "copy_links": null, "delete": false, "dest": "/tmp/2", "dest_port": 22, "dirs": false, "existing_only": false, "group": null, "links": null, "mode": "push", "owner": null, "partial": false, "perms": null, "private_key": null, "recursive": null, "rsync_opts": null, "rsync_path": null, "rsync_timeout": 0, "set_remote_user": true, "src": "/tmp/1", "ssh_args": null, "times": null, "verify_host": false}}, "msg": "rsync: link_stat \"/tmp/1\" failed: No such file or directory (2)\nrsync error: some files/attrs were not transferred (see previous errors) (code 23) at main.c(1183) [sender=3.1.1]\n", "rc": 23}
```
##### Actual Results:

```
TASK [synchronize] *************************************************************
task path: /home/dromanov/ansible-test/deploy.yml:5
ESTABLISH LOCAL CONNECTION FOR USER: dromanov
localhost EXEC /bin/sh -c '( umask 22 && mkdir -p "$( echo $HOME/.ansible/tmp/ansible-tmp-1453373528.13-196644905147624 )" && echo "$( echo $HOME/.ansible/tmp/ansible-tmp-1453373528.13-196644905147624 )" )'
localhost PUT /tmp/tmp1TrJyg TO /home/dromanov/.ansible/tmp/ansible-tmp-1453373528.13-196644905147624/synchronize
localhost EXEC /bin/sh -c 'LANG=ru_RU.UTF-8 LC_ALL=ru_RU.UTF-8 LC_MESSAGES=ru_RU.UTF-8 /usr/bin/python /home/dromanov/.ansible/tmp/ansible-tmp-1453373528.13-196644905147624/synchronize; rm -rf "/home/dromanov/.ansible/tmp/ansible-tmp-1453373528.13-196644905147624/" > /dev/null 2>&1'
An exception occurred during task execution. The full traceback is:
Traceback (most recent call last):
  File "/home/dromanov/ansible/lib/ansible/executor/process/worker.py", line 114, in run
    self._shared_loader_obj,
  File "/home/dromanov/ansible/lib/ansible/executor/task_executor.py", line 119, in run
    res = self._execute()
  File "/home/dromanov/ansible/lib/ansible/executor/task_executor.py", line 410, in _execute
    result = self._handler.run(task_vars=variables)
  File "/home/dromanov/ansible/lib/ansible/plugins/action/synchronize.py", line 248, in run
    if 'SyntaxError' in result['msg']:
KeyError: 'msg'

fatal: [localhost]: FAILED! => {"failed": true, "stdout": ""}
```

**Comment by dmromanov** [comment]
Additionally reproduced on Ansible 2.0.0.2

**Comment by abadger** [comment]
I pushed a fix for this as part of my update to the synchronize module to fix it for 2.0.1.  The commit is in devel and stable-2.0 branches.
